addon/components/x-toggle/component.js

```diff
@@ -1,30 +1,71 @@
 import Ember from 'ember';
 import layout from './template';
 
-const { on, run, computed, observer } = Ember;
+const { run, computed } = Ember;
+const a = Ember.A;
 
-export default Ember.Component.extend({
+const xToggle = Ember.Component.extend({
   layout: layout,
-  tagName: 'span',
-  classNameBindings: ['toggled:x-toggle-container-checked', 'disabled:x-toggle-container-disabled'],
-  classNames: ['x-toggle-container'],
-  disabled: false,
-  value: false,
-  toggled: false,
+  tagName: '',
+
   name: 'default',
+  disabled: false,
+  value: 'off',
+  offLabel: 'Off:off',
+  onLabel: 'On:on',
+
+  init() {
+    this._super(...arguments);
+    run.schedule('afterRender', () => {
+      // if value is not set to a valid state suggest a default to the container
+      const {toggled, _onValue, _offValue} = this.getProperties('toggled', '_onValue', '_offValue');
+
+      if(toggled === undefined) {
+        const response = this.ddau('onToggle', {
```

Comment:
are we sure we want to fire actions when the component is created? It looks like there is an issue about that https://github.com/knownasilya/ember-cli-toggle/issues/44